lib/roby/actions/models/action.rb

```diff
@@ -198,7 +198,11 @@ def run(action_interface, arguments = Hash.new)
                     if !arguments.empty?
                         raise ArgumentError, "#{name} expects no arguments, but #{arguments.size} are given"
                     end
-                    result = action_interface.send(name).as_plan
+                    result = action_interface.send(name)
+                    if(!result.class.has_ancestor?(returned_type))
```

Comment:
Actually, the #as_plan is really needed as e.g. syskit returns an object that's not a task (but object#as_plan turns it into one).

Why don't you check on the result of #as_plan ?